src/inference/src/ie_core.cpp

```diff
@@ -929,6 +935,7 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         auto deviceName = pluginName;
         if (deviceName == ov::DEFAULT_DEVICE_NAME)
             deviceName = "AUTO";
+        stripDeviceName(deviceName, "-");
```

Comment:
from I understand of this feature, AUTO:-GPU will run success even if  there is no GPU device on the machine,
so GetCPPPluginByName("-GPU") should not be call instead of  remove "-" in GetCPPPluginByName

Reply:
1) is this feature targeting -GPU also? from the jira and requirement, I understand only need to disable CPU usage to save CPU, or disable CPU accelerator, a.k.a cpu FIL
2) for  the mentioned case above, if hw does not have GPU, or user don't want to use GPU, why don't user just use AUTO:CPU or AUTO, or AUTO: CPU, MYRAID, I think maybe it's different scenario from the jira described.

Reply:
Clarify the expectation as below:

- App set excluded device only like AUTO:-CPU

> - AUTO query available devices and create default device list
> - AUTO remove excluded device from device list
> - AUTO do selection logic with updated device list

- App set candidate devices and excluded device like AUTO:GPU,-iGPU

> - AUTO parse candidate and create device list include GPU.0, GPU.1 and etc
> - AUTO remove excluded device from device list. excluded device may be part of device full name
> - AUTO do selection logic with updated device list

- Name of exclude device can be set of devices like -GPU or part of device full name like -iGPU

Reply:
![image](https://user-images.githubusercontent.com/32348500/163547201-691df374-db8b-45a7-b13e-a2fab3e6af2f.png)
@wangleis  please check the impl again, and attached some test results above if this is the expected result, if yes, I will enable optimize and cases based on it.

Reply:
cumulative cases, below all with -hint ctput:
AUTO: -CPU -> MULTI:GPU.1,GPU.0
AUTO:-GPU.0 ->MULTI:GPU.1,CPU

Reply:
Update expectation in description for 2022.2 release. Supporting full device name is not in scope.

Reply:
> cumulative cases, below all with -hint ctput: AUTO: -CPU -> MULTI:GPU.1,GPU.0 AUTO:-GPU.0 ->MULTI:GPU.1,CPU

@songbell Yes. it is our expectation for this PR in 2022.2 release.

Reply:
@tiger100256-hu  can you help a review again?

Reply:
I have no other question